CMakeLists.txt

```diff
@@ -43,7 +47,7 @@ if (NOT DEFINED CMAKE_INSTALL_LIBDIR)
 	include(GNUInstallDirs)
 endif()
 
-set (SRT_VERSION 1.4.2)
+# Sets SRT_VERSION_MJOR, SRT_VERSION_MINOR, SRT_VERSION_PATCH
```

Comment:
This looks pointless according to the documentation of the `PROJECT` command: https://cmake.org/cmake/help/v3.0/command/project.html

According to this description you should simply call
```
project(SRT VERSION 1.4.2)
```
and done. This command will set appropriately `SRT_VERSION`, `SRT_VERSION_MAJOR`, `SRT_VERSION_MINOR` and `SRT_VERSION_PATCH` - that is, the same that the below `set_version_variables` is trying to do.

As per the `CMP0058` policy, yes, it should be set to NEW before `project` command, so I think it would be better off this way:

1. First include `haiUtil` to have access to `set_version_variables` function (also set CMAKE_MODULE_PATH as above it).
2. Set the SRT_VERSION variable (for common use)
3. Condition for cmake version_less 3.0.0
4. For version >3.0.0, set the policy to new and use `project(SRT VERSION ${SRT_VERSION})`
5. For version <3.0.0, use `project(SRT)`, then call `set_version_variables` as below.

(LANGUAGES is optional and defaults to "C CXX", so I doubt it needs to be specified explicitly).

Reply:
If you are patient enough to play around more, you can enclose the above procedure in a macro, place it in `haiUtil.cmake`, and call it, for example, `srt_project_version` so that all you'll have will be
```
set (CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/scripts")
include(haiUtil)

srt_project_version(1.4.2)

```

Reply:
> This command will set appropriately SRT_VERSION, SRT_VERSION_MAJOR, SRT_VERSION_MINOR and SRT_VERSION_PATCH - that is, the same that the below set_version_variables is trying to do.

I think it will set `PROJECT_VERSION_MAJOR` etc. instead of `SRT_VERSION_MINOR`.
Maybe we should switch to this naming.
[See PROJECT_VERSION_MAJOR](https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION_MAJOR.html#variable:PROJECT_VERSION_MAJOR).

Reply:
My assumption was wrong, it sets both `PROJECT_VERSION_MAJOR` and `SRT_VERSION_MAJOR`:
```
The project() command stores the version number and its components in variables

PROJECT_VERSION, <PROJECT-NAME>_VERSION
PROJECT_VERSION_MAJOR, <PROJECT-NAME>_VERSION_MAJOR
PROJECT_VERSION_MINOR, <PROJECT-NAME>_VERSION_MINOR
PROJECT_VERSION_PATCH, <PROJECT-NAME>_VERSION_PATCH
PROJECT_VERSION_TWEAK, <PROJECT-NAME>_VERSION_TWEAK
```